src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -48,6 +48,29 @@ class UploadingJobsDisplay extends Component {
     this.uploadJobs();
   }
 
+  componentDidUpdate() {
+    if (!this.props.files) {
+      return;
+    }
+
+    let dontLeave = false;
+
+    for (const file in this.props.files) {
+      if (this.props.files[file].status.includes('Uploading')) {
+        dontLeave = true;
+        break;
+      }
+    }
+    // prevent from leaving the page in case of download in progress
+    if (dontLeave) {
+      window.onbeforeunload = () => {
+        return false;
+      };
+    } else {
+      window.onbeforeunload = null;
```

Comment:
This reset is also needed in `componentWillUnmount`, as this feature should only work on this page. Please create the method for this like `resetLeaveThePagePrevention`.

```suggestion
  resetLeaveThePagePrevention() {
    window.onbeforeunload = null;
  }
```

Reply:
Done